crates/primitives/src/header.rs

```diff
@@ -177,6 +177,18 @@ impl Header {
         Ok(())
     }
 
+    /// Checks if the block's timestamp is in the past compared to the parent block's timestamp.
+    pub fn is_timestamp_in_past(&self, parent_timestamp: u64) -> bool {
+        self.timestamp <= parent_timestamp
+    }
+
+    /// Checks if the block's timestamp is in the future based on the present timestamp.
+    ///
+    /// Clock can drift but this can be consensus issue.
+    pub fn is_timestamp_in_future(&self, present_timestamp: u64) -> bool {
```

Comment:
I think these should mention that these are only relevant pre-merge

and I think we should rename `is_timestamp_in_future` because this check is a bit confusing because `is_timestamp_in_future == true` is considered an invalid block: `ConsensusError::TimestampIsInFuture`

maybe smth like `exceeds_allowed_future_timestamp`